include/effects/matrix/PatternSMSupernova.h

```diff
@@ -4,60 +4,80 @@
 
 // Inspired by https://editor.soulmatelights.com/gallery/1923-supernova
 
-class PatternSMSupernova : public LEDStripEffect 
-{
-    void Init()
-    {
-        memset(enlargedObjectTime, 0, sizeof(enlargedObjectTime));
-        memset(PosX, 0, sizeof(PosX));
-        memset(PosY, 0, sizeof(PosY));
-        memset(Hue, 0, sizeof(Hue));
-        memset(State, 0, sizeof(State));
-        memset(SpeedX, 0, sizeof(SpeedX));
-        memset(SpeedY, 0, sizeof(SpeedY));
-        memset(IsShift, 0, sizeof(IsShift));
+// A more cache-friendly version of the 8 independent arrays that were
+// used. This allows better locality per member, especially since we
+// access them sequentially.
+// Nothing special here; just default-zero initialized struct.
+class Object {
+  public:
+    void Objects() {
+        PosX = 0.0f;
+        PosY = 0.0f;
+        State = 0.0f;
+        SpeedX = 0.0f;
+        SpeedY = 0.0f;
+        Hue = 0;
+        IsShift = false;
     }
```

Comment:
What exactly is the purpose of this function? It looks like a constructor, except the function name Object**s** doesn't match the class name Object. As far as I can tell, it also isn't used anywhere.